xscen/extract.py

```diff
@@ -798,6 +806,15 @@ def search_data_catalogs(
     if restrict_members is not None and len(catalogs) > 0:
         catalogs = _restrict_multimembers(catalogs, id_columns, restrict_members)
 
```

Comment:
Is there a reason to do this on `catalogs` instead of doing it way earlier on `catalog` (after `if exclusions`, for example)? The way this is currently implemented, you potentially lose a lot of time searching through simulations that you might end up deleting anyway.

At the very least, this should come before `restrict_resolution` and `restrict_members`, since these should be the last checkups (aka you've done all your work, and only then do you eliminate duplicates if some still exist).

Reply:
There is no reason, I just figured it was the best way to exclude stuff as this is what had been done for other restrictions. I can put it after `if exclusion`.